config/channel/consolidated/resources/kafka-channel.yaml

```diff
@@ -24,52 +24,58 @@ metadata:
 spec:
   group: messaging.knative.dev
   preserveUnknownFields: false
-  validation:
-    openAPIV3Schema:
-      type: object
-        # this is a work around so we don't need to flush out the
-        # schema for each version at this time
-        #
-        # see issue: https://github.com/knative/serving/issues/912
-      x-kubernetes-preserve-unknown-fields: true
   names:
     kind: KafkaChannel
     plural: kafkachannels
     singular: kafkachannel
     categories:
-    - all
-    - knative
-    - messaging
-    - channel
+      - all
+      - knative
+      - messaging
+      - channel
     shortNames:
-    - kc
+      - kc
   scope: Namespaced
-  subresources:
-    status: {}
-  additionalPrinterColumns:
-    - name: Ready
-      type: string
-      JSONPath: ".status.conditions[?(@.type==\"Ready\")].status"
-    - name: Reason
-      type: string
-      JSONPath: ".status.conditions[?(@.type==\"Ready\")].reason"
-    - name: URL
-      type: string
-      JSONPath: .status.address.url
-    - name: Age
-      type: date
-      JSONPath: .metadata.creationTimestamp
   versions:
-  - name: v1alpha1
-    served: true
-    storage: true
-  - name: v1beta1
-    served: true
-    storage: false
+    - &version
+      name: v1alpha1
+      served: true
+      storage: true
+      subresources:
+        status: { }
+      schema:
+        openAPIV3Schema: &openAPIV3Schema
+          type: object
+          # this is a work around so we don't need to flush out the
+          # schema for each version at this time
+          #
+          # see issue: https://github.com/knative/serving/issues/912
+          x-kubernetes-preserve-unknown-fields: true
+      additionalPrinterColumns:
+        - name: Ready
+          type: string
+          jsonPath: ".status.conditions[?(@.type==\"Ready\")].status"
+        - name: Reason
+          type: string
+          jsonPath: ".status.conditions[?(@.type==\"Ready\")].reason"
+        - name: URL
+          type: string
+          jsonPath: .status.address.url
+        - name: Age
+          type: date
+          jsonPath: .metadata.creationTimestamp
+    - <<: *version
+      name: v1beta1
+      served: true
+      storage: false
+      schema:
+        openAPIV3Schema:
+          << : *openAPIV3Schema
   conversion:
     strategy: Webhook
-    conversionReviewVersions: ["v1beta1", "v1alpha1"]
-    webhookClientConfig:
-      service:
-        name: kafka-webhook
-        namespace: knative-eventing
+    webhook:
+      conversionReviewVersions: [ "v1", "v1beta1" ]
```

Comment:
@pierDipi 
Please only change the CRD to use Kube's v1 ApiExtensions. 

Bumping the version of `messaging.knative.dev` to `v1` is a separate work.

Reply:
This refers to the conversion request version.

Reply:
From: https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definition-versioning/
```
# conversionReviewVersions indicates what ConversionReview versions are understood/preferred by the webhook.
# The first version in the list understood by the API server is sent to the webhook.
# The webhook must respond with a ConversionReview object in the same version it received.
```

Reply:
Ok, so, not related to KafkaChannel's version (messaging.knative.dev)?

Reply:
Yes, it's unrelated, see https://github.com/knative/eventing/blob/master/config/core/resources/eventtype.yaml, `EventType` is in `v1alpha1` and `v1beta1` and the conversionReviewVersions is set to [`[ "v1", "v1beta1" ]`](https://github.com/knative/eventing/blob/dccb17e10b6bc04e51f0d6f1a871f4676fe89e2f/config/core/resources/eventtype.yaml#L158)

Reply:
ha, ok.

Reply:
Isn't that the conversion of the CRD api itself ?

Reply:
Nope, they are versions of the conversion request and response, see https://github.com/kubernetes/apiextensions-apiserver/blob/1fa7e1a40da83d381ce598c1dc4095debc040c60/pkg/apis/apiextensions/v1/types.go#L434-L476 and https://github.com/kubernetes/apiextensions-apiserver/blob/1fa7e1a40da83d381ce598c1dc4095debc040c60/pkg/apis/apiextensions/v1beta1/types.go#L475-L522

`v1alpha1` version has never been created, (first CRDs release branch) https://github.com/kubernetes/apiextensions-apiserver/tree/release-1.7/pkg/apis/apiextensions